test/RelayHub.test.ts

```diff
@@ -976,6 +982,377 @@ describe('RelayHub', function () {
+          data = swap.interface.encodeFunctionData('claim', [
+            constants.HashZero,
+            ethers.utils.parseEther('2'),
+            constants.AddressZero,
+            500,
+          ]);
+
+          const deployRequest = cloneDeployRequest({
+            request: {
+              index: nextWalletIndex.toString(),
+              to: swap.address,
+              data,
+              tokenContract: constants.AddressZero,
+            },
+            relayData: {
+              callForwarder: minimalBoltzFactory.address,
+            },
+          });
+
+          const { signature } = await signEnvelopingRequest(
+            deployRequest,
+            owner
+          );
+
+          nextWalletIndex++;
+
+          const deployCall = relayHub
+            .connect(relayWorker)
+            .deployCall(deployRequest, signature, { gasLimit });
+
+          await expect(deployCall).to.be.rejectedWith(
+            'Could not transfer Ether'
+          );
+        });
+
+        it('should fail if not enough native token to pay', async function () {
+          const deployRequest = cloneDeployRequest({
+            request: {
+              index: nextWalletIndex.toString(),
+              to: swap.address,
+              data,
+              tokenAmount: claimedValue.add(ethers.utils.parseEther('0.5')),
+              tokenContract: constants.AddressZero,
+            },
+            relayData: {
+              callForwarder: minimalBoltzFactory.address,
+            },
+          });
+
+          const { signature } = await signEnvelopingRequest(
+            deployRequest,
+            owner
+          );
+
+          nextWalletIndex++;
+
+          const deployCall = relayHub
+            .connect(relayWorker)
+            .deployCall(deployRequest, signature, { gasLimit });
+
+          await expect(deployCall).to.be.rejectedWith(
+            'Unable to pay for deployment'
+          );
+        });
+
+        it('should succeed paying with native token', async function () {
+          const deployRequest = cloneDeployRequest({
+            request: {
+              index: nextWalletIndex.toString(),
+              to: swap.address,
+              data,
+              tokenContract: constants.AddressZero,
+              tokenAmount: ethers.utils.parseEther('0.01'),
+            },
+            relayData: {
+              callForwarder: minimalBoltzFactory.address,
+            },
+          });
+
+          const { signature } = await signEnvelopingRequest(
+            deployRequest,
+            owner
+          );
+
+          nextWalletIndex++;
+
+          const deployCall = relayHub
+            .connect(relayWorker)
+            .deployCall(deployRequest, signature, { gasLimit });
+
+          await expect(deployCall).not.to.be.rejected;
+        });
+
+        it.skip('should fail if not enough gas to pay for native transfer', async function () {
```

Comment:
why are we skipping this test?

Reply:
Is this related to the automatic token gas set by the VM? If so, I'd either change the test not to expect a fail and adding a comment to describe why we don't expect this to fail

Reply:
it is related to that, let me fix it.

Reply:
updated